### PR TITLE
minor: if build is triggered by PR, use PR branch, otherwise use master

### PR DESCRIPTION
**Pull request by daniilyar**

### CODE REVIEW

**Comment by daniilyar** [comment]
Work-in-progress - please don't merge this
I need to update this PR few times to understand how does the Jenkins PR builder job works with it when PR is triggered from fork

File: Jenkinsfile [hunk 1]
```diff
@@ -79,24 +84,39 @@ pipeline {
...
+
+        echo GREEN("Branch: $GIT_BRANCH")
+
+        // Print the build envvars
+        echo sh(returnStdout: true, script: 'env')
+      }
+    }
+
+    stage ("Prepare (triggered by PR)") {
+      when { expression { IS_TRIGGERED_FOR_PR } }
```
Comment (romani) [inline Jenkinsfile]: for => by

you use by in all other cases.

Comment (daniilyar) [reply Jenkinsfile]: Fixed

**Comment by daniilyar** [comment]
@romani here is how the Jenkins PR build output looks now: http://jenkins.checkstyle.org/job/pr-builder-checkstyle/29/console

In Jenkins output, I added some useful info such as link to PR, PR author and commit title

**Comment by daniilyar** [comment]
Jenkins, test this please

**Comment by daniilyar** [comment]
Travis build failed with some `curl: (7) Failed to connect to rpmfind.net port 443: No route to host` error. I am updating the PR so it restarts the Travis. I hope that on the third retry it will run to success :)

**Comment by rnveach** [comment]
@daniilyar We are seeing this error everywhere, so it is unrelated to your PR.

**Comment by romani** [comment]
@daniilyar , exactly this build item  failed after merge due to missed update in whitelist. I fixed master. Please try to run locally failed commands to confirm that it pass.

**Comment by daniilyar** [comment]
@romani , sorry, I don't understand. What failing commands?

**Comment by romani** [comment]
all that CIs do , is possible to relaunch on local. This is critical point for us, as some CI might be damaged at any time , but this should not block the process.

command is `./.ci/test-spelling-unknown-words.sh`.

**Comment by daniilyar** [comment]
> possible to relaunch on local.
@romani, of course, it is good, but possible still doesn't mean convenient :) the current quality of CI runner scripts does not allow any contributir to 'just run' them on typical system, such as Mac, Win, Fedora/RedHat/CentOS Lunix - only some Ubuntu versions are supported.

For example, on my clean system, the spellchecker script prints 1k+ lines patch and proposes to add to whitelist words that are already there. See full output on gists: https://gist.github.com/daniilyar/8d0746cf387e8b7b0d797f3fea47a750

I will check why it fails tomorrow.

**Comment by romani** [comment]
> Will check why it does that tomorrow.

finally you started to share problems ... good. We need to make steps to be working in other environments too. Please create issue on this, eventually we will find a solution.
Simple example from my experience: It took us several years to finally make simple "mvn test" pass on all environments and all languages. As we become to know a problem, and collect details .... we will find a solution. ALL will benefit.

**Comment by daniilyar** [comment]
@romani, no need to try supporting all possible contributor platforms. Let's do Vagrant or Docker instead :)

One of example of inconvenience is that most ci scripts are designed for execution from git repo root folder and fail from anywhere else. E.g.:
```
.ci/test-spelling-unknown-words.sh - works
cd .ci && ./test-spelling-unknown-words.sh - fails
xxx/checkstyle/.ci/test-spelling-unknown-words.sh - fails
```

Second is that all CI scripts lack proper qutation around path arguments with makes them fail on any catalog with spaces in directory name.

That is all because the CI scripts were simply designed for running on CI where paths and folder names are strict and properly defined once and forever, and not for running locally by contributor )

So we need to first 'redesign' CI scripts a bit. I would start from updating bash scripts to conform to common Bash best practices such as "use relative paths", "quote everything" and "anything goes to the function".

I will fix these issues that I see and some other small things soon in separate PRs.

**Pull request merged**